assets/src/infrastructure/ui/navigation/pagination/PerPage.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+
+import { Select, SelectProps } from '@infraUI/inputs';
+import { PerPageProps } from './types';
+
+const calculatePageNumber = (newPerPage: number, prevPerPage: number, total: number): number => {
+	const perPage = typeof newPerPage === 'undefined' ? prevPerPage : newPerPage;
+	return Math.floor((total - 1) / perPage) + 1;
+};
+
+const PerPage: React.FC<PerPageProps> = ({ locale, onChangePerPage, pageNumber, perPage, perPageOptions, total }) => {
```

Comment:
I tried to find this component here: https://github.com/react-component/pagination/blob/master/examples/jumper.js
and here: https://github.com/ant-design/ant-design/blob/master/components/pagination/Pagination.tsx, but this not found it.
It was written from scratch?

Reply:
Yes, read the PR description.